[bgfx] Add new port

Outurnate opened the pull request:
Rewrite of #9163

I've only written a port for bgfx.  There's no way to get GENie to use the artifacts from another port (i.e. bx or bimg), so I saw little point in exposing them

There was a bug in vcpkg_build_make as well - the docs specified that there was a `TARGET` parameter where no such parameter existed.  I added this parameter, as well as two others to override the target for debug/release builds

Outurnate committed:
[bgfx] Add new port

Outurnate committed:
[bgfx] remove debug messages

JackBoosY commented:
Seems have problem:
```
make: *** No rule to make target 'vs2017-debug32'.  Stop.
```

Outurnate committed:
added WORKING_DIRECTORY option to vcpkg_build_make

Outurnate committed:
[bgfx] use msbuild directly on windows

Outurnate committed:
removed merge markers

Outurnate committed:
[bgfx] correct platform names for msbuild

Outurnate commented:
A couple more changes in `vcpkg_build_make`.  I added a new `WORKING_DIRECTORY` option.  I removed the section that closes `gpg-agent.exe` on Windows since it doesn't work (the code was copy/pasted from the icu port - it still had a ref to a variable that was only in the original port).  It seemed unnecessary since the process is a part of Msys2 and in no way prevents the build from completing

As it turns out, the windows make target for the project is sort of broken.  The makefile just calls `devenv PATH_TO_SLN /Build` which ends up just launching Visual Studio.  I used the makefile to generate the SLN file, then ran msbuild over that

JackBoosY commented:
/azp run

JackBoosY commented:
/azp run

JackBoosY commented:
/azp run

JackBoosY commented:
/azp run

JackBoosY commented:
Only have windows build failure, please download failure log [here](https://dev.azure.com/vcpkg/public/_build/results?buildId=25475&view=artifacts&type=publishedArtifacts) and resolve them.

Thanks.

Outurnate commented:
There's nothing in any of those log bundles.  I tested this port on the Windows 10 dev VM provided by Microsoft

JackBoosY commented:
@Outurnate I will provide your failure logs tomorrow.

Commit:
[bgfx] Fix build failure on Windows

JackBoosY commented:
@Outurnate I think you catch a vcpkg bug.

JackBoosY commented on ports/bgfx/portfile.cmake:
```diff
@@ -0,0 +1,84 @@
+
+if(VCPKG_TARGET_IS_WINDOWS)
+  set(MAKE_TARGET_PLATFORM vs2017)
+  set(EXECUTABLE_SUFFIX ".exe")
```
Could you remove it and replace it to `VCPKG_TARGET_EXECUTABLE_SUFFIX`?

LilyWangL commented:
@Outurnate Can you address the review suggestions? Thanks!

StuartDAdams commented:
Would be really eager to see this library added to vcpkg.

JackBoosY commented:
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.
